Added ecs-anywhere role and anisble-galaxy collection files

sergio-gaggle opened the pull request:
Added ecs-anywhere role and tasks files. Also added manifest file for ansible-galaxy collection packaging.

sergio-gaggle committed:
Added ecs-anywhere role and anisble-galaxy collection files

sergio-gaggle committed:
Updated auth check to run once

cplee commented on MANIFEST.json:
```diff
@@ -0,0 +1,23 @@
+  ],
+  "description": "ECS Anywhere Collection",
+  "license": [],
+  "license_file": "LICENSE",
```
`LICENSE` file is missing - please add an MIT license file to the repo

sergio-gaggle replied:
Added MIT license file and wrote modules to remove dependency on the AWS CLI. aws_ssm_activation module creates/gets/deletes SSM activations and aws_ssm_instance module gets/de-registers instances from SSM

cplee commented on galaxy.yml:
```diff
@@ -0,0 +1,57 @@
+# Either a single license or a list of licenses for content inside of a collection. Ansible Galaxy currently only
+# accepts L(SPDX,https://spdx.org/licenses/) licenses. This key is mutually exclusive with 'license_file'
+license:
+- GPL-2.0-or-later
```
no GPL please...let's use MIT

sergio-gaggle replied:
Updated to MIT

cplee commented on roles/ecs_anywhere/meta/main.yml:
```diff
@@ -0,0 +1,23 @@
+  descpription: Ansible role that configures ecs anywhere instance
+  company: Gaggle Net
+  issue_tracker_url: https://github.com/gaggle-net/ansible-ecs-anywhere-playbook/issues
+  license: Apache-2.0
```
change to MIT license for consistency

sergio-gaggle replied:
Updated to MIT

cplee commented on README.md:
````diff
@@ -1 +1,44 @@
+  become: yes
+```
+
+## Running the Playbook
````
Would be good to document the parameters that are `required` and `optional`

sergio-gaggle replied:
Added section to README describing which variables are optional or required.

cplee commented on roles/ecs_anywhere/tasks/main.yml:
```diff
@@ -0,0 +1,38 @@
+
+- name: ECS Anywhere | Download ECS Anywhere Install Script
+  get_url: 
+    url: https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh
```
the bucket name should come from `aws_bucket_name` parameter. Also, shouldn't use `preview` anymore as per: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

cplee commented on roles/ecs_anywhere/defaults/main.yml:
```diff
@@ -0,0 +1,6 @@
+aws_region: us-west-2
+aws_dir: /opt/amazon/ecs-anywhere
+aws_ecs_cluster: default
+aws_s3_bucket: amazon-ecs-agent-packages-preview
```
shouldn't use `preview` anymore as per: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

cplee commented on roles/ecs_anywhere/tasks/main.yml:
```diff
@@ -0,0 +1,38 @@
+  become: yes
+
+- name: ECS Anywhere | Create SSM Activation Code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ aws_ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
`create-activation` is called in 3 separate places - can this be simplified?

cplee commented on roles/ecs_anywhere/tasks/register-ssm-instance.yml:
```diff
@@ -0,0 +1,46 @@
+    state: absent
+
+- name: ECS Anywhere | Create SSM Activation Code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ aws_ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
`create-activation` is called in 3 separate places - can this be simplified?

cplee commented on roles/ecs_anywhere/tasks/aws-script-install.yml:
```diff
@@ -0,0 +1,39 @@
+  become: yes
+
+- name: create ssm activation code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
`create-activation` is called in 3 separate places - can this be simplified?

cplee commented on roles/ecs_anywhere/tasks/aws-script-install.yml:
```diff
@@ -0,0 +1,39 @@
+    url: https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh
+    dest: "{{ aws_dir }}"
+    mode: '0550'
+    checksum: sha256:https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh.sha256
```
these are the old buckets from preview: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

cplee reviewed (changes requested):
Can we make `aws_profile` optional? We use profiles a lot internally, but that isn't the best assumption for others...some may not use profiles the way we do.

Also, there is a lot of dependency on the AWS CLI - any reason why we couldn't use Ansible plugins and write the code in Python using boto3 to remove the need for the AWS CLI?

Commit:
ecs_anywhere: Added ssm plugins [SA-9185]
Added plugins for creating ssm activations and deregistering
ssm instances

Commit:
ecs_anywhere: updated playbooks to include local plugin [SA-9185]
Updated playbook and tasks to use local aws_ssm plugins instead
of using shell commands

Commit:
ecs_anywhere: added warnings [SA-9185]
Added replaced failures with warnings when activation
or instance is not found.

Commit:
ecs_anywhere: updated tasks to use local plugins [SA-9185]
Updated tasks to use local plugins instead of using the aws
cli. Updated url for aws anywhere install script to no
longer use preview version.

sergio-gaggle committed:
ecs_anywhere: Renamed playbooks dir and added examples [SA-9185]
Renamed example_playbooks dir to playbooks and updated examples

sergio-gaggle committed:
ecs_anywhere: Updated license info [SA-9185]
Updated license to use MIT and updated README

sergio-gaggle committed:
ecs_anywhere: Added MIT License [SA-9185]
Added MIT license file and updated manifest

cplee committed:
Update main.yml

update license

cplee commented:
@sergio-gaggle - this is looking REALLY good 🤩

Only 1 last request before we merge it - can you update the `README.md` to include a list of vars like:

| variable | description | required? | default |
| -------- | ----------- | ---------- | ------- | 
| iam_role | Which IAM role to associate with instance | yes | - |
| aws_ecs_cluster | Name of ECS cluster to use | yes | - |
| aws_region | Which region to use for this ECS instance | no | `us-west-2` |
| aws_install_dir | Where to install ECS anywhere | no | `/opt/amazon/ecs-anywhere` |
| aws_script_url | Location of install script | no | `https://amazon-ecs-agent.s3.amazonaws.com/ecs-anywhere-install-latest.sh` |

Commit:
ecs_anywhere: updated aws script name [SA-9185]
Updated aws install script name to include "latest"

Commit:
Merge branch 'SA-9185' of https://github.com/gaggle-net/ansible-collection-ecs-anywhere into SA-9185
Updated aws install script name

tmz-gaggle commented:
Nice work putting this together Sergio!

Is Fedora really a supported OS for the collection?  ECS Anywhere itself says versions 32 and 33 are supported, which means they're  a month or two away from not supporting any current Fedora releases.  Fedora 32 was EOL in May.  Fedora 33 will be EOL 4 weeks after Fedora 35 is released, which is scheduled for October 19 or 26.  The versions in the collection here are 27 and 28, which have been EOL since November 2018 and May 2019, respectively.

It might be worth removing Fedora entirely unless we want to test it on Fedora 33 and update the README and role metadata?

Not the most useful of reviews, I know.  I just noticed those old versions in the README and thought I'd ask.

Commit:
ecs_anywhere: added vars descriptions [SA-9185]
Added vars decsriptions and requirements to README

Commit:
ecs_anywhere: removed fedora references from README [SA-9185]
removed references to fedora per Tod's review. Will add back
once confirmed collection works on fedora

cplee commented:
@sergio-gaggle - ready to merge?

sergio-gaggle commented:
@cplee yup, ready to merge to master
